src/Evolution/Systems/Cce/Actions/BoundaryComputeAndSendToEvolution.hpp

```diff
@@ -72,5 +108,87 @@ struct BoundaryComputeAndSendToEvolution {
         true);
   }
 };
+
+/*!
+ * \ingroup ActionsGroup
+ * \brief Submits a request for CCE boundary data at the specified `time` to the
+ * `Cce::GhWorldtubeInterfaceManager`, and sends to the `EvolutionComponent`
+ * (template argument) it is ready.
+ *
+ * \details This uses the `Cce::GhWorldtubeInterfaceManager` to perform all of
+ * the work of managing the buffer of data sent from the Gh system and
+ * interpolating if necessary and supported. This dispatches then to
+ * `Cce::Actions::SendToEvolution<GhWorldtubeBoundary<Metavariables>,
+ * EvolutionComponent>` if the boundary data is ready, otherwise
+ * simply submitting the request and waiting for data to become available via
+ * `Cce::Actions::ReceiveGhWorldtubeData`, which will call
+ * `Cce::Actions::SendToEvolution<GhWorldtubeBoundary<Metavariables>,
+ * EvolutionComponent>` as soon as the data becomes available.
+ *
+ * \ref DataBoxGroup changes:
+ * - Adds: nothing
+ * - Removes: nothing
+ * - Modifies:
+ *   - `Tags::GhInterfaceManager`
+ */
+template <typename Metavariables, typename EvolutionComponent>
+struct BoundaryComputeAndSendToEvolution<GhWorldtubeBoundary<Metavariables>,
+                                         EvolutionComponent> {
+  template <typename ParallelComponent, typename... DbTags, typename ArrayIndex,
+            Requires<tmpl2::flat_any_v<cpp17::is_same_v<
+                ::Tags::Variables<
+                    typename Metavariables::cce_boundary_communication_tags>,
+                DbTags>...>> = nullptr>
+  static void apply(db::DataBox<tmpl::list<DbTags...>>& box,
+                    Parallel::ConstGlobalCache<Metavariables>& cache,
+                    const ArrayIndex& /*array_index*/,
+                    const TimeStepId& time) noexcept {
+    db::mutate<Tags::GhInterfaceManager>(
+        make_not_null(&box),
+        [&time, &
+         cache ](const gsl::not_null<db::item_type<Tags::GhInterfaceManager>*>
+                     interface_manager) noexcept {
+          (*interface_manager)->request_gh_data(time);
+          if (const auto gh_data =
```

Comment:
move the assignment out of the `if()`, and then `static_cast<bool>` the result in the `if()` to make it clear that the validity of the data is being tested